### PR TITLE
#167700724 Refactoring code and middlewares

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Refactoring code and middlewares

#### Description of Task to be completed?
Restructure folder structure
Refactored response messages
Refactored isAdmin middleware
Refactored validation middleware
Moved JWT middleware

#### How should this be manually tested?
Git clone this repo and run `npm install && npm test`

#### What are the relevant pivotal tracker stories?
#167700724

### CODE REVIEW

File: server/v1/models/bookingModel.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+class Booking {
+  /**
+  * class constructor
+  * @param data
+  */
+  constructor() {
+    this.bookings = [
+      {
+        id: 1,
+        trip_id: 2,
```
Comment (Kasulejoseph) [inline server/v1/models/bookingModel.js]: Don't hardcode the booking object.

File: server/v1/models/tripModel.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+class Trip {
+  /**
+  * class constructor
+  * @param {object} data
+  */
+  constructor() {
+    this.trips = [
+      {
+        id: 1,
+        seating_capacity: 45,
```
Comment (Kasulejoseph) [inline server/v1/models/tripModel.js]: Avoid hardcoding data/content in the models

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  res.header('Access-Control-Allow-Origin', '*'); // Access to any client
+  res.header('Access-Control-Allow-Headers', '*');
+  if (req.method === 'OPTIONS') {
+    res.header('Access-Control-Allow-Methods', 'PUT, POST, PATCH DELETE, GET');
+    res.status(200).json({});
+  }
+  next();
+});
+
+app.get('/', (req, res) => res.status(200).send({ message: "Bienvenue, this is WayFarer's!!!" }));
```
Comment (Kasulejoseph) [inline server/server.js]: This should be in the routers file.

Comment (nignanthomas) [reply server/server.js]: Done

File: server/v1/models/tripModel.js (hunk 2 above)
Comment (nignanthomas) [reply server/v1/models/tripModel.js]: Done

File: server/v1/models/bookingModel.js (hunk 1 above)
Comment (nignanthomas) [reply server/v1/models/bookingModel.js]: Done

File: server/v1/middlewares/validationMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const route = req.route.path;
+  const method = req.method.toLowerCase();
+
+  if (validMethods.includes(method) && validators[route]) {
+    const schema = validators[route];
+
+    return Joi.validate(req.body, schema, (error, data) => {
+      if (error) {
+        console.log('Validation Failed');
+        console.log(error.details[0].message);
```
[the remark this answers is not in the document]
Comment (nignanthomas) [reply server/v1/middlewares/validationMiddleware.js]: Done

File: server/v1/helpers/passport.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+const JwtStrategy = require('passport-jwt').Strategy;
+
+// JWT Strategy
+passport.use(new JwtStrategy({
+  jwtFromRequest: ExtractJwt.fromHeader('authorization'),
+  secretOrKey: JWT_SECRET,
+  // eslint-disable-next-line consistent-return
+}, async (payload, done) => {
+  try {
+    // Find the user specified in token
```
[the remark this answers is not in the document]
Comment (nignanthomas) [reply server/v1/helpers/passport.js]: Done.
Thanks for the advice

File: server/v1/helpers/idValidator.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import Joi from '@hapi/joi';
+
+const tripIdValidator = (data) => {
+  const schema = Joi.object().keys({
+    tripId: Joi.number().integer().required(),
+  });
+  return Joi.validate(data, schema);
+};
+const bookingIdValidator = (data) => {
```
[the remark this answers is not in the document]
Comment (nignanthomas) [reply server/v1/helpers/idValidator.js]: In progress...

File: server/v1/config/index.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  JWT_SECRET: 'wayfarerauthentication',
```
[the remark this answers is not in the document]
Comment (nignanthomas) [reply server/v1/config/index.js]: In progress...

**Pull request merged**